### PR TITLE
Add HilbertCurve implementation

### PR DESCRIPTION
**Pull request by sandan**
Implemented HilbertCurve in scala using the UG library and the toy interface for space filling curves. The HilbertCurve uses the CoordinateWGS84 class to encapsulate Doubles representing latitude and longitude.

Additionally I created a separate sub project for the toy interface in the project/Build.scala file, the project compiles on my machine.

### CODE REVIEW

File: project/Build.scala [hunk 1]
```diff
@@ -46,10 +47,32 @@ object Build extends Build {
         "-language:existentials",
         "-feature"),
       libraryDependencies ++= Seq(
-        "org.scalatest" %% "scalatest" % "2.2.4" % "test"
+        "org.scalatest" %% "scalatest" % "2.2.4" % "test",
+        "com.google.uzaygezen" % "uzaygezen-core" % "0.2" ,
+        "org.apache.accumulo" % "accumulo-core" % "1.6.0"
```
Comment (jnh5y) [inline project/Build.scala]: If this isn't used, we'll hold off for now.

**Comment by bretts61** [comment]
Had a no element found error when trying to benchmark hilbert curves. Line 98 in HilbertCurve.scala
screen shot 2015-03-20 at 9 40 57 am

My code was:
val sfc = HilbertCurve(i)
val range = sfc.RangeQuery(-178.123456, -86.398493, 179.3211113, 87.393483)

**Comment by jnh5y** [comment]
Can you guys change the code on/around line 98 in HilbertCurve.scala logs a quick message if the iterator is empty?

**Comment by lossyrob** [comment]
I'm going to merge this in as this PR is getting big, and the progress is being made. Let's aim for small PR's to be merged in, so we can all collaborate on the master branch. I'll try to be more attentive to reviewing and merging.

**Pull request merged**